docs/3.0-migration-guide.d/remove_ssl_record_checking.md

```diff
@@ -0,0 +1,13 @@
+Remove MBEDTLS_SSL_RECORD_CHECKING option and enable its action by default
+--------------------------------------------------------------------------
+
+This change does not affects users who use the default config.h, as the
+option MBEDTLS_SSL_RECORD_CHECKING was already on by default.
+
+This option was added only to controls compilation of one function
+(mbedtls_ssl_check_record()) used in DTLS to check a buffer's validity and
+authenticity. Switching it off poses a security risk.
```

Comment:
Sorry, that's not correct, there is no security risk in switching the option off (otherwise we would have put a big warning about it in `config.h`!). The purpose of the function is to check the record _without consuming it_ - otherwise (and prior to this function being introduced) the record's validity and authenticity are always checked when the application attempts to consume the record.

Suggested text: "
This option was added only to controls compilation of one function, `mbedtls_ssl_check_record()`, which is only useful in some specific cases, so it was made optional to allow people who don't need it to save some code size. However, the same effect can be achieve by using link-time garbage collection.

For users who changed the default setting of the option, you just need to change your config/build system to remove that change."

Reply:
Done. Thanks Manuel!